src/App.tsx

```diff
@@ -21,75 +22,96 @@ enum SortType {
   LENGTH,
 }
 
-type ReorderOptions = {
-  sortType: SortType,
+type State = {
   isReversed: boolean,
+  sortType: number,
 };
 
-// Use this function in the render method to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+export class App extends React.Component<{}, State> {
+  goods = [...goodsFromServer];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  state = {
+    isReversed: false,
+    sortType: 0,
+  };
 
-  return visibleGoods;
-}
+  buttonClickReverse = (): void => {
+    this.goods = this.goods.reverse();
+    this.setState(prevState => (
+      { ...prevState, isReversed: !prevState.isReversed }));
+  };
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
+  buttonClickAbc = (): void => {
+    this.goods = [...this.goods.sort((a, b) => a.localeCompare(b))];
+    this.setState(prevState => ({ ...prevState, sortType: SortType.ALPHABET }));
+  };
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+  buttonClickLength = (): void => {
+    this.goods = [...this.goods.sort((a, b) => a.length - b.length)];
+    this.setState(prevState => ({ ...prevState, sortType: SortType.LENGTH }));
+  };
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+  buttonClickReset = (): void => {
+    this.goods = [...goodsFromServer];
+    this.setState(prevState => (
+      { ...prevState, isReversed: false, sortType: SortType.NONE }));
+  };
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+  render() {
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            id="SortByAbc"
+            className={`button is-info
+              ${(this.state.sortType === SortType.ALPHABET ? '' : 'is-light')}`}
```

Comment:
what for? same code without useless module.

Reply:
because you will not write like if .... - add '' , what a reason to do so?
You can have more conditions in future and sense to use classnames  will increase

Reply:
when I obtain more than 1 extra class in the future - then I use react classname module, and now it's not needed.

Reply:
it's good practice train yourself to use the right tools right away

Reply:
I have enough practice, save this for yourself.